### PR TITLE
xds: fix xdsClient resource not exist for invalid resource, fix xdsServerWrapper start on resource not exist

### PR DESCRIPTION
**Pull request by YifeiZhuang**
https://github.com/grpc/grpc-java/issues/8657

cc. @sanjaypujare

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/ClientXdsClient.java [hunk 1]
```diff
@@ -2214,11 +2214,13 @@ private void handleResourceUpdate(
             }
             retainedResources.add(edsName);
           }
-          continue;
+        } else if (invalidResources.contains(resourceName)) {
```
Comment (ejona86) [inline xds/src/main/java/io/grpc/xds/ClientXdsClient.java:2217]: This looks incomplete, as the case when `type != ResourceType.LDS && type != ResourceType.CDS` will no longer trigger `onAbsent()`, ever.

**Review by ejona86 — COMMENTED** [review]
Drive-by mini-review

File: xds/src/main/java/io/grpc/xds/ClientXdsClient.java (hunk 1 above)
Comment (YifeiZhuang) [reply xds/src/main/java/io/grpc/xds/ClientXdsClient.java:2217]: Yea I was confused as well it was existing behavior for other resource types. Looks a bug?
cc. @dapengzhang0

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/ClientXdsClient.java:2217]: For RDS,EDS `onAbsent()` is handled below  (`dependentSubscribers`)

https://github.com/grpc/grpc-java/pull/8660/files#diff-6b019eb1d3518381d336168847890513a235ea4974fcb3c76a2e9ace43f1d460R2233

@YifeiZhuang your change is only for LDS and CDS type.

Comment (YifeiZhuang) [reply xds/src/main/java/io/grpc/xds/ClientXdsClient.java:2217]: > This looks incomplete, as the case when type != ResourceType.LDS && type != ResourceType.CDS will no longer trigger onAbsent(), ever.

@ejona86  was your original question about [RDS and EDS](https://github.com/grpc/grpc-java/blob/a5c526c12f699af9a846d9faabc18038ef431675/xds/src/main/java/io/grpc/xds/ClientXdsClient.java#L1413) never gets `onAbsent()`? if so, then it looks by design and makes sense as Penn explained to me offline. They are SoW not incremental. 
Then I think the change is what we want.

Comment (ejona86) [reply xds/src/main/java/io/grpc/xds/ClientXdsClient.java:2217]: Oh, I see it. I mixed up the indentation.

**Review by dapengzhang0 — APPROVED** [review]
LGTM

File: xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java [hunk 2]
```diff
@@ -240,6 +248,9 @@ public void run() {
       fail("Start should throw exception");
     } catch (ExecutionException ex) {
       assertThat(ex.getCause()).isInstanceOf(IOException.class);
+      fail("Start should not fail");
+    } catch (TimeoutException ex) {
+      assertThat(start.isDone()).isFalse();
```
Comment (dapengzhang0) [inline xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java:253]: Isn't the exception deterministic?

Comment (dapengzhang0) [reply xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java:253]: Seems should expect only one exception.

**Review by dapengzhang0 — CHANGES_REQUESTED** [review]

File: xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java (hunk 2 above)
Comment (YifeiZhuang) [reply xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java:250]: yea!

**Review by dapengzhang0 — APPROVED** [review]

**Pull request merged**